src/chainparamsbase.cpp

```diff
@@ -23,6 +24,9 @@ void SetupChainParamsBaseOptions(ArgsManager& argsman)
     argsman.AddArg("-segwitheight=<n>", "Set the activation height of segwit. -1 to disable. (regtest-only)", ArgsManager::ALLOW_ANY | ArgsManager::DEBUG_ONLY, OptionsCategory::DEBUG_TEST);
     argsman.AddArg("-testnet", "Use the test chain. Equivalent to -chain=test.", ArgsManager::ALLOW_ANY, OptionsCategory::CHAINPARAMS);
     argsman.AddArg("-vbparams=deployment:start:end", "Use given start/end times for specified version bits deployment (regtest-only)", ArgsManager::ALLOW_ANY | ArgsManager::DEBUG_ONLY, OptionsCategory::CHAINPARAMS);
+    argsman.AddArg("-signet", "Use the signet chain. Note that the network is defined by the signet_challenge parameter", ArgsManager::ALLOW_ANY, OptionsCategory::CHAINPARAMS);
+    argsman.AddArg("-signet_challenge", "Blocks must satisfy the given script to be considered valid (only for signet networks; defaults to the global default signet test network challenge)", ArgsManager::ALLOW_STRING, OptionsCategory::CHAINPARAMS);
```

Comment:
In commit "add signet chain and accompanying parameters":

There doesn't seem to be a default signet test network challenge in this commit.

Reply:
No, the default signet stuff is explicitly added in a later commit. The code could, in theory, be merged without that commit to start off with a no-default signet merge. Doesn't look like anyone is opposed to the default we have though, but still.

Reply:
BTW: this is the first command line argument with ` _`, or any kind of separator—for that matter— in it. We might want to discuss what we like to see here. Personally, I'd expect `-signet-challenge`. Or just `-signetchallenge` to be consistent with say, `-segwitheight`.

Reply:
Very vaguely related, I just saw this: https://twitter.com/koenvervloesem/status/1305911670083067904

Reply:
I don't have a strong opinion on it, other than it feels like a good idea to clearly separate signet stuff from the rest. I believe we will have several more parameters in the future, e.g. for selecting soft forks or to define alternative challenges (opt-in reorg chain, if we end up going that route).

Reply:
Sure, agree on that, but starting the options with `signet` should already be enough to keep them separate from the rest.

Reply:
Switching to `-signet<suffix>`.

Reply:
Also addressed original note that there is no default signet at this point; this line is now modified in the "hard coded default signet" commit to add in the note about default network.